namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -46,20 +47,33 @@ private[consul] object SvcAddr {
     preferServiceAddress: Option[Boolean] = None,
     tagWeights: Map[String, Double] = Map.empty,
     stats: Stats,
-    stateWatch: v1.PollState[String, v1.IndexedServiceNodes]
+    stateWatch: v1.PollState[String, v1.IndexedServiceNodes],
+    transferMetaData: Boolean = false
   )(implicit timer: Timer = DefaultTimer): InstrumentedVar[Addr] = {
     val meta = mkMeta(key, datacenter, domain)
-    def getAddresses(index: Option[String]): Future[v1.Indexed[Set[Address]]] = {
+
+    def getAddresses(index: Option[String]): Future[v1.Indexed[Addr.Bound]] = {
       val apiCall = consulApi.serviceNodes(
         key.name,
         datacenter = Some(datacenter),
         tag = key.tag,
         blockingIndex = index,
-        consistency = consistency,
-        retry = false
+        consistency = consistency
       )
       v1.InstrumentedApiCall.execute(apiCall, stateWatch)
-        .map(indexedToAddresses(preferServiceAddress, tagWeights))
+        .map {
+          case v1.Indexed(nodes, idx) => {
+            // we use the metadata from the first node, because all nodes from
+            // the same service are assumed to have identical metadata.
+            // For reference: https://www.consul.io/api/catalog.html
+            val meta: v1.Metadata = nodes.headOption.flatMap(_.ServiceMeta) -> transferMetaData match {
```

Comment:
Take it or leave it (TIOLI) suggestion:

This might be clearer as:

```
val meta = nodes.headOption.flatMap(_.ServiceMeta) match {
  case Some(serviceMeta) if transferMetadata => serviceMeta
  case _ => Map.empty
}
```